wire/test/serializinglocalbus.go

```diff
@@ -0,0 +1,42 @@
+package test
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"perun.network/go-perun/wire"
+)
+
+// SerializingLocalBus is a local bus that also serializes messages for testing.
+type SerializingLocalBus struct {
+	*wire.LocalBus
+	t *testing.T
+}
+
+// NewSerializingLocalBus creates a new serializing local bus.
+func NewSerializingLocalBus(t *testing.T) *SerializingLocalBus {
+	return &SerializingLocalBus{
+		LocalBus: wire.NewLocalBus(),
+		t:        t,
+	}
+}
+
+// Publish publishes the message on the bus.
+func (b *SerializingLocalBus) Publish(ctx context.Context, e *wire.Envelope) (err error) {
+	var buf bytes.Buffer
+	err = e.Encode(&buf)
+	if err != nil {
+		return
+	}
+
+	var _e wire.Envelope
+	err = _e.Decode(&buf)
+	if err != nil {
+		return
+	}
+
+	wire.TestMsg(b.t, e.Msg)
```

Comment:
I think either En/decoding manually or calling this test should be enough.

Reply:
This covers two different cases. There could be an error in the implementation of `TestMsg`, so encoding and decoding is useful. On the other hand, encoding and decoding will not give more details about where the encoding may have failed.

Reply:
If you suspect `TestMsg` to have a bug in its implementation, it is better to write a proper test for `TestMsg` (can use the `pkg/test.Tester` for this) and then use it with full trust. This is what helpers like `TestMsg` are for, to avoid duplication.

Reply:
I don't really suspect it to have a bug, but who knows. Anyways, I will keep the wire.TestMsg and remove the encoding decoding above.